### PR TITLE
Feat/server title support

### PR DESCRIPTION
**Pull request by peknur**
Add support for server title field

### CODE REVIEW

File: upcloud/resource_upcloud_server.go [hunk 1]
```diff
@@ -400,7 +409,11 @@ func resourceUpCloudServerRead(ctx context.Context, d *schema.ResourceData, meta
 		return diag.FromErr(err)
 	}
 	_ = d.Set("hostname", server.Hostname)
-	_ = d.Set("title", server.Title)
+	if server.Title != cloudServerDefaultTitleFromHostname(server.Hostname) {
+		_ = d.Set("title", server.Title)
+	} else {
+		_ = d.Set("title", nil)
```
Comment (Darep) [inline upcloud/resource_upcloud_server.go:415]: What's the effect of setting this to nil here? Will TF make sure it's the default one?

File: upcloud/resource_upcloud_server.go [hunk 2]
```diff
@@ -740,3 +759,11 @@ func resourceUpCloudServerDelete(ctx context.Context, d *schema.ResourceData, me
 
 	return diags
 }
+
+func cloudServerDefaultTitleFromHostname(hostname string) string {
+	suffix := " (managed by terraform)"
+	if len(hostname)+len(suffix) > cloudServerTitleLength {
+		hostname = fmt.Sprintf("%s...", hostname[:cloudServerTitleLength-len(suffix)-3])
```
Comment (Darep) [inline upcloud/resource_upcloud_server.go:766]: UpCloud API accepts the single character ellipsis `…`
Could save two characters by using that here 😉

File: upcloud/resource_upcloud_server.go [hunk 3]
```diff
@@ -20,6 +20,8 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
 )
 
+var cloudServerTitleLength int = 64
```
Comment (Darep) [inline upcloud/resource_upcloud_server.go:23]: should this be `const`?

File: upcloud/resource_upcloud_server.go [hunk 4]
```diff
@@ -740,3 +759,11 @@ func resourceUpCloudServerDelete(ctx context.Context, d *schema.ResourceData, me
 
 	return diags
 }
+
+func cloudServerDefaultTitleFromHostname(hostname string) string {
+	suffix := " (managed by terraform)"
```
Comment (Darep) [inline upcloud/resource_upcloud_server.go:764]: Could this be a `const` above as well(?). Or is there a problem with placing the same const object in a bunch of strings 🤔

**Review by Darep — APPROVED** [review]
Did not test locally, but the changes look good to me! Note that I'm not a Go/Terraform expert 😉 

Dropped some questions/notes, check those maybe :blush:

File: upcloud/resource_upcloud_server.go (hunk 1 above)
Comment (ka-myl) [reply upcloud/resource_upcloud_server.go:415]: This sets the `title` value in `tfstate` to empty string. And I'm not sure if this is actually the desired behaviour. `title` field is marked as computed, so if I don't provide one, I would expect the computed value to end up in my tfstate.

**Review by ka-myl — COMMENTED** [review]
I don't think we need a separate acceptance test for this feature, but perhaps it would be good to add title (and checks for it) to the `TestUpcloudServer_basic`?

**Pull request merged**